flink-ml-lib/src/main/java/org/apache/flink/ml/feature/vectorassembler/VectorAssembler.java

```diff
@@ -74,38 +76,65 @@ public Table[] transform(Table... inputs) {
         return new Table[] {outputTable};
     }
 
-    private static class AssemblerFunc implements FlatMapFunction<Row, Row> {
+    private static class AssemblerFunction implements FlatMapFunction<Row, Row> {
         private final String[] inputCols;
         private final String handleInvalid;
+        private final int[] sizeArray;
 
-        public AssemblerFunc(String[] inputCols, String handleInvalid) {
+        public AssemblerFunction(String[] inputCols, String handleInvalid, int[] sizeArray) {
             this.inputCols = inputCols;
             this.handleInvalid = handleInvalid;
+            this.sizeArray = sizeArray;
         }
 
         @Override
         public void flatMap(Row value, Collector<Row> out) {
             int nnz = 0;
             int vectorSize = 0;
             try {
-                for (String inputCol : inputCols) {
+                for (int i = 0; i < inputCols.length; ++i) {
+                    String inputCol = inputCols[i];
                     Object object = value.getField(inputCol);
                     Preconditions.checkNotNull(object, "Input column value should not be null.");
                     if (object instanceof Number) {
+                        Preconditions.checkArgument(
+                                sizeArray[i] == 1,
+                                "Inconsistent vector size, setSize is "
+                                        + sizeArray[i]
+                                        + ", but current size is "
+                                        + 1
+                                        + ".");
                         nnz += 1;
-                        vectorSize += 1;
+                        vectorSize += sizeArray[i];
                     } else if (object instanceof SparseVector) {
+                        int localSize = ((SparseVector) object).size();
+                        Preconditions.checkArgument(
+                                sizeArray[i] == localSize,
+                                "Inconsistent vector size, setSize is "
+                                        + sizeArray[i]
+                                        + ", but current vector size is "
+                                        + localSize
+                                        + ".");
                         nnz += ((SparseVector) object).indices.length;
-                        vectorSize += ((SparseVector) object).size();
+                        vectorSize += sizeArray[i];
                     } else if (object instanceof DenseVector) {
+                        int localSize = ((DenseVector) object).size();
+                        Preconditions.checkArgument(
+                                sizeArray[i] == localSize,
+                                "Inconsistent vector size, setSize is "
+                                        + sizeArray[i]
+                                        + ", but current vector size is "
+                                        + localSize
```

Comment:
The three `checkArgument`s are almost the same and we may merge them into one common function.